contrib/clojure-package/test/org/apache/clojure_mxnet/resource_scope_test.clj

```diff
@@ -0,0 +1,65 @@
+;;
+;; Licensed to the Apache Software Foundation (ASF) under one or more
+;; contributor license agreements.  See the NOTICE file distributed with
+;; this work for additional information regarding copyright ownership.
+;; The ASF licenses this file to You under the Apache License, Version 2.0
+;; (the "License"); you may not use this file except in compliance with
+;; the License.  You may obtain a copy of the License at
+;;
+;;    http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+
+(ns org.apache.clojure-mxnet.resource-scope-test
+  (:require [org.apache.clojure-mxnet.ndarray :as ndarray]
+            [org.apache.clojure-mxnet.symbol :as sym]
+            [org.apache.clojure-mxnet.resource-scope :as resource-scope]
+            [clojure.test :refer :all]))
+
+(deftest test-resource-scope-with-ndarray
+  (let [x (ndarray/ones [2 2])
+        return-val (resource-scope/using
+                    (def temp-x (ndarray/ones [3 1]))
+                    (def temp-y (ndarray/ones [3 1]))
+                    (let [z {:just-a-test (def temp-z (ndarray/ones [3 3]))}
```

Comment:
Is there a more natural use case that the nested `def` forms that still illustrates the behavior and is amenable to testing?

Reply:
I agree the def are weird - do you think an atom with a map of ndarrays under test that could be `swap` ed in would be better?

Reply:
Hmm. I'd say yes _cautiously_. Tradeoff IMO: it would look a bit more idiomatic, but also put more indirection in the tests.

Reply:
Thanks for the feedback and review @benkamphaus - I updated the test. They are an improvement. I feel like it could be further improved as well, but I can't think of an immediate way :)